anipy_cli/url_handler.py

```diff
@@ -113,74 +116,74 @@ def pick_range(self):
+
+        return self.entry
+
 
 class videourl():
-    """ 
+    """
     Class that fetches embed and
-    stream url. 
+    stream url.
     """
+
     def __init__(self, entry, quality) -> None:
-        self.entry = entry 
+        self.entry = entry
         self.qual = quality
         self.session = requests.Session()
         retry = Retry(connect=3, backoff_factor=0.5)
         adapter = HTTPAdapter(max_retries=retry)
         self.session.mount('http://', adapter)
         self.session.mount('https://', adapter)
         self.ajax_url = "https://gogoplay.io/encrypt-ajax.php"
-        self.key = '3235373436353338353932393338333936373634363632383739383333323838'
-        self.iv = '34323036393133333738303038313335'
-    
+
+        self.mode = AES.MODE_CBC
+        self.size = AES.block_size
+        self.pad = lambda s: s + (self.size - len(s) %
+                                  self.size) * chr(self.size - len(s) % self.size)
+        self.iv = binascii.unhexlify(
+            bytes('34323036393133333738303038313335'.encode("utf-8")))
+        self.key = binascii.unhexlify(bytes(
+            "3235373436353338353932393338333936373634363632383739383333323838".encode("utf-8")))
+
     def get_entry(self):
         """
         Returns the entry with stream and emebed url fields filled
         which was previously passed to this class.
         """
         return self.entry
 
-    def embed_url(self):    
+    def embed_url(self):
         r = self.session.get(self.entry.ep_url)
         response_err(r, self.entry.ep_url)
         soup = BeautifulSoup(r.content, "html.parser")
         link = soup.find("a", {"href": "#", "rel": "100"})
         loc_err(link, self.entry.ep_url, "embed-url")
         self.entry.embed_url = f'https:{link["data-video"]}'
-    
-    def decrypt_link(self):    
+
+    def decrypt_link(self):
         splt_url = urlsplit(self.entry.embed_url)
         video_id = parse_qs(splt_url.query)['id'][0]
-        # no permanent solution
-        cmd = f'printf \"{video_id}\" | openssl enc -aes-256-cbc -nosalt -e -K \"{self.key}\" -iv \"{self.iv}\" -a'
-        return sp.getoutput(cmd)
-        
-        # the code where I gave up, leaving this here for the future
-
-        #BLOCK_SIZE = 16
-        #pad = lambda s: s + (BLOCK_SIZE - len(s) % BLOCK_SIZE) * chr(BLOCK_SIZE - len(s) % BLOCK_SIZE)
-        #
-        #private_key = hashlib.sha256(self.key.encode("utf-8")).digest()
-        #iv = binascii.unhexlify(self.iv) 
-        #video_id = pad(video_id).encode('utf-8')
-        #cipher = AES.new(private_key, AES.MODE_CBC, iv)
-        #print(base64.b64encode(cipher.encrypt(video_id)))
-        #ajax = base64.b64encode(iv + cipher.encrypt(video_id))
+
+        cryptor = AES.new(self.key, self.mode, self.iv)
+        encrypted = cryptor.encrypt(
+            bytearray(self.pad(video_id+"\n"), "utf-8"))
+        ajax = base64.b64encode(encrypted)
+
+        return str(ajax).replace("'", "")[1:]
```

Comment:
You can use `ajax.decode('utf-8')` instead of `return str(ajax).replace("'", "")[1:]`